web/src/settings_ui.ts

```diff
@@ -22,20 +22,29 @@ export const strings = {
 // Generic function for informing users about changes to the settings
 // UI.  Intended to replace the old system that was built around
 // direct calls to `ui_report`.
+
+interface DoSettingsChangeOptions {
+    success_msg_html?: string;
+    failure_msg_html?: string;
+    success_continuation?: (data: unknown) => void;
+    error_continuation?: (xhr: JQuery.jqXHR) => void;
+    sticky?: boolean;
+    $error_msg_element?: JQuery<HTMLElement>;
+}
```

Comment:
As noted in #24759, we would prefer `type DoSettingsChangeOptions` for consistency. Could you also reason why you make them all optional? Some of them do have default values in `do_settings_change`, so the implementation wouldn't expect `undefined` values for those.